NitroxServer/ConsoleCommands/UpgradeCommand.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.IO;
+using System.Threading;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.Helper;
+using NitroxModel.Server;
+using NitroxServer.ConsoleCommands.Abstract;
+using NitroxServer.Serialization;
+using NitroxServer.Serialization.Upgrade;
+using NitroxServer.Serialization.World;
+
+namespace NitroxServer.ConsoleCommands
+{
+    internal sealed class UpgradeCommand : Command
+    {
+        private readonly WorldPersistence worldPersistence;
+        private readonly ServerConfig serverConfig;
+        private readonly SaveDataUpgrade[] upgrades;
+        private readonly Server server;
+
+        public UpgradeCommand(WorldPersistence worldPersistence, ServerConfig serverConfig, SaveDataUpgrade[] upgrades, Server server) : base("upgrade", Perms.CONSOLE, "Upgrades the save file to the next version")
+        {
+            this.worldPersistence = worldPersistence;
+            this.serverConfig = serverConfig;
+            this.upgrades = upgrades;
+            this.server = server;
+        }
+
+        protected override void Execute(CallArgs args)
+        {
+            string saveDir = serverConfig.SaveName;
+            string fileEnding = worldPersistence.SaveDataSerializer.GetFileEnding();
+            SaveFileVersion saveFileVersion = worldPersistence.SaveDataSerializer.Deserialize<SaveFileVersion>(Path.Combine(saveDir, "Version" + fileEnding));
+
+
+            if (saveFileVersion.Version == NitroxEnvironment.Version)
+            {
+                SendMessage(args.Sender, "Save files are already at the newest version");
+            }
+            else if (serverConfig.SerializerMode == ServerSerializerMode.PROTOBUF)
+            {
+                SendMessage(args.Sender, "Can't upgrade while using ProtoBuf as serializer");
+            }
+            else
+            {
+                foreach (SaveDataUpgrade upgrade in upgrades)
+                {
+                    if (upgrade.TargetVersion > saveFileVersion.Version)
+                    {
+                        SendMessage(args.Sender, $"Applying {upgrade.GetType().Name}");
+                        upgrade.UpgradeData(saveDir, fileEnding);
+                    }
+                }
+                worldPersistence.SaveDataSerializer.Serialize(Path.Combine(saveDir, "Version" + fileEnding), new SaveFileVersion());
+                SendMessage(args.Sender, $"Save file was successfully upgraded to {NitroxEnvironment.Version}");
+                server.Stop(false);
+                Thread.Sleep(2000);
```

Comment:
Why sleep?

Reply:
That way the last message is readable. Without it the server is closed immediately which gives the impression of a fatal error.

Reply:
Make sense. Though I recommend adding a feature to make the server stop with a message to display which will:
1. Show message in console.
2. Wait for user to enter any key to stop.

Reply:
Wouldnt that be implementable via adding some method to NitroxLogger
```csharp
void LogAndWait()
{
  Info("Press any key to continue");
  Console.Read();
}
```

Reply:
I don't agree with that because you don't always have a console when logging. Would need to do something similar to `InGameLogger` abstraction to noop if no console is available.

Reply:
I would add the method to NitroxServer/Program